lib/index.js

```diff
@@ -273,6 +273,48 @@ exports.createStorage = async (ctx) => {
     }
 };
 
+exports.deleteStorage = async (ctx) => {
+    debug("DELETE STORAGE");
+
+    // Find the Client
+    const issuerId = `uri:fusebit-template:${ctx.functionId}:${ctx.body.subscriptionId}:${ctx.body.boundaryId}:${ctx.body.functionId}`;
+    const subject = 'client-1';
+    debug('Looking up Client ID', issuerId);
+    const response = await Superagent.get(`${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/client?issuerId=${
+            encodeURIComponent(issuerId)
+        }&subject=${subject}&include=all`)
+        .set("Authorization", ctx.headers['authorization']); // pass-through authorization
+    const client = response.body.items && response.body.items[0];
+    debug('Found client', client);
+
+    if (client) {
+        // Delete the storage
+        const allow = (client.access && client.access.allow || []);
+        for (var i = 0; i < allow.length; i++) {
+            if (allow[i].action.indexOf('storage:') === 0) {
+                await Superagent.delete(`${ctx.body.baseUrl}/v1${allow[i].resource}`)
+                    .set("Authorization", ctx.headers['authorization']) // pass-through authorization
+                    .ok(r => r.status < 300 || r.status === 404)
+                debug('Deleted storage', allow[i]);
+            }
+        }
+        // Delete the client
+        debug('Deleting client');
+        await Superagent.delete(`${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/client/${client.id}`)
+            .set("Authorization", ctx.headers['authorization']) // pass-through authorization
```

Comment:
Not sure what we return when the client does not exist - in general all the logic in this function should likely ignore 404s.

Reply:
I made them all throw unless it's a 2xx or 404